src/main/java/ch/njol/skript/expressions/ExprOnlinePlayersCount.java

```diff
@@ -0,0 +1,146 @@
+@Since("INSERT VERSION")
+public class ExprOnlinePlayersCount extends SimpleExpression<Number> {
+
+	static {
+		Skript.registerExpression(ExprOnlinePlayersCount.class, Number.class, ExpressionType.PROPERTY,
+				"[the] [(1¦(real|default)|(2¦fake|shown|displayed))] online players (count|amount|number|size)",
+				"[the] [(1¦(real|default)|(2¦fake|shown|displayed))] (count|amount|number|size) of online players");
+	}
+
+	@SuppressWarnings("null")
+	private Kleenean delay;
+
+	private boolean isReal;
+
+	@SuppressWarnings({"unchecked", "null"})
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, SkriptParser.ParseResult parseResult) {
+		boolean isServerPingEvent = ScriptLoader.isCurrentEvent(ServerListPingEvent.class);
+		boolean isPaperEvent = Skript.classExists("com.destroystokyo.paper.event.server.PaperServerListPingEvent") && ScriptLoader.isCurrentEvent(PaperServerListPingEvent.class);
+		if (parseResult.mark == 2) {
+			if (isServerPingEvent) {
+				Skript.error("The 'shown' online players count expression requires PaperSpigot 1.12.2+");
+				return false;
+			} else if (!isPaperEvent) {
+				Skript.error("The 'shown' online players count expression can't be used outside of a server list ping event" + (isServerPingEvent ? " and requires PaperSpigot 1.12.2+" : ""));
+				return false;
+			}
+		}
+		isReal = (parseResult.mark == 0 && !isPaperEvent) || parseResult.mark == 1;
+		delay = isDelayed;
+		return true;
+	}
+
+	@Override
+	public boolean isSingle() {
+		return true;
+	}
+
+	@Override
+	@Nullable
+	public Number[] get(Event e) {
+		if (isReal)
+			return CollectionUtils.array(PlayerUtils.getOnlinePlayers().size());
+		else
+			return CollectionUtils.array(((PaperServerListPingEvent) e).getNumPlayers());
+	}
+
+	@Override
+	@Nullable
+	public Class<?>[] acceptChange(Changer.ChangeMode mode) {
+		if (!isReal) {
+			if (delay == Kleenean.TRUE) {
+				Skript.error("Can't change the shown online players count anymore after the server list ping event has already passed");
+				return null;
+			}
+			if (mode == Changer.ChangeMode.SET ||
+					mode == Changer.ChangeMode.ADD ||
+					mode == Changer.ChangeMode.REMOVE ||
+					mode == Changer.ChangeMode.DELETE ||
+					mode == Changer.ChangeMode.RESET)
+				return CollectionUtils.array(Number.class);
+		}
+		return null;
+	}
+
+	@Override
+	@SuppressWarnings("null")
+	public void change(Event e, @Nullable Object[] delta, Changer.ChangeMode mode) {
+		PaperServerListPingEvent event = (PaperServerListPingEvent) e;
+		switch (mode){
+			case SET:
+				event.setNumPlayers(((Number) delta[0]).intValue());
+				break;
+			case ADD:
+				event.setNumPlayers(event.getNumPlayers() + ((Number) delta[0]).intValue());
+				break;
+			case REMOVE:
+				event.setNumPlayers(event.getNumPlayers() - ((Number) delta[0]).intValue());
+				break;
+			case DELETE:
+			case RESET:
+				event.setNumPlayers(PlayerUtils.getOnlinePlayers().size());
+				break;
```

Comment:
What's your view on only allowing `RESET` here?

Reply:
I agreed this but i just changed my opinion, because `delete the fake online players count` makes sense.

Reply:
Check which syntax is used then 👍